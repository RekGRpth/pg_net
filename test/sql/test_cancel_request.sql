begin;

    with req as (
        select net.async_get('net://supabase.io') as id
    )
    select
        net.cancel_request(id)
    from
        req;

    select
        id, url
    from
        net.request_queue;

rollback;
