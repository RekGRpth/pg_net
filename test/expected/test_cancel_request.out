begin;
    with req as (
        select http.async_get('http://supabase.io') as id
    )
    select
        http.cancel_request(id)
    from
        req;
 cancel_request 
----------------
              2
(1 row)

    select
        *
    from
        http.request_queue;
 id | url | params | headers | timeout_seconds | curl_opts | is_completed | delete_after 
----+-----+--------+---------+-----------------+-----------+--------------+--------------
(0 rows)

rollback;
